app/content/webviewPreload.js

```diff
@@ -2,255 +2,377 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-var messages = require('../../js/constants/messages')
-var KeyCodes = require('../../js/constants/keyCodes')
-
-var ipc = {
-  events: {},
-  embedder: null,
-  processMessage: function (event) {
-    if (event.origin === 'file://') {
-      this.embedder = event.source
-      var cb = this.events[event.data[0]]
-      cb && cb.apply(null, event.data)
-    }
-  },
-  on: function (name, cb) {
-    this.events[name] = cb
-  },
-  send: function () {
-    var args = Array.prototype.slice.call(arguments)
-    this.embedder && this.embedder.postMessage(args, 'file://')
+// inject missing DOM Level 3 KeyEvent
+// https://www.w3.org/TR/2001/WD-DOM-Level-3-Events-20010410/DOM3-Events.html#events-Events-KeyEvent
+if (typeof KeyEvent === 'undefined') {
+  var KeyEvent = {
```

Comment:
I intentionally left it out. It's part of DOM Level 3 and exists in Firefox, but Chrome didn't implement it so we might as well make it available